src/filter.jl

```diff
@@ -1,144 +1,134 @@
 
-    # Arguments:
-    - `R_S::T`: filter radius
-    - `kv`: wavevector, tuple of (kx, ky, kz)
+"""
+Computes the Fourier representation of a Gaussian filter with width ``R_S``.
+You would multiply this with the FFT'd field and then IFFT to apply the filter.
+
+# Arguments:
+- `R_S::T`: filter radius
+- `kv`: wavevector, tuple of (kx, ky, kz)
 
-    # Returns:
-    - `Array{T,3}`: fourier representation of gaussian filter
-    """
+# Returns:
+- `Array{T,3}`: fourier representation of gaussian filter
+"""
+function kspace_gaussian_filter(R_S::T, kv) where T
     kx, ky, kz = kv
     nx, ny, nz = length(kx), length(ky), length(kz)
     filter_k = zeros(T, (nx, ny, nz))
-    for ix = 1:nx, iy = 1:ny, iz = 1:nz
-        filter_k[ix, iy, iz] = exp(
-            -(kx[ix]^2 + ky[iy]^2 + kz[iz]^2) * R_S^2 / 2)
+    Threads.@threads for ix = 1:nx
+        for iy = 1:ny, iz = 1:nz
+            filter_k[ix, iy, iz] = exp(
+                -(kx[ix]^2 + ky[iy]^2 + kz[iz]^2) * R_S^2 / 2)
+        end
     end
     return filter_k
 end
 
+
+"""
+Apply a gaussian filter to a density field with smoothing radius `R_S`
+making use of wavevectors `kv`
+
+# Arguments:
+- `f::AbstractArray{T,3}`: density field
+- `R_S::T`: filter radius
+- `kv`: wavevector, tuple of (kx, ky, kz)
+
+# Returns:
+- Array{T,3}: smoothed density field
+"""
 function smooth_gauss(f::AbstractArray{T,3}, R_S::T, kv) where T
-    """
-    Apply a gaussian filter to a density field with smoothing radius `R_S` 
-    making use of wavevectors `kv`
-    
-    # Arguments:
-    - `f::AbstractArray{T,3}`: density field
-    - `R_S::T`: filter radius
-    - `kv`: wavevector, tuple of (kx, ky, kz)
-    
-    # Returns:
-    - Array{T,3}: smoothed density field
-    """
+
     GF = kspace_gaussian_filter(R_S, kv)  # get filter in Fourier space
     f_Rn = (real(ifft(GF .* fft(f))))
     f_Rn = f_Rn .* (sum(f)/sum(f_Rn))  # normalize
     return f_Rn
 end
 
-function smooth_loggauss(f::AbstractArray{T,3}, R_S::T, kv) where T
-    """
-    Apply a gaussian filter to a density field with smoothing radius `R_S` 
-    making use of wavevectors `kv` using log smoothing
-    
-    # Arguments:
-    - `f::AbstractArray{T,3}`: density field
-    - `R_S::T`: filter radius
-    - `kv`: wavevector, tuple of (kx, ky, kz)
-    
-    # Returns:
-    - Array{T,3}: log-smoothed density field
-    """
-    GF = kspace_gaussian_filter(R_S, kv)  # get filter in Fourier space
-    f_Rn = 10 .^(
-        real(ifft(GF .* fft(log10.(f)))))
-    f_Rn = f_Rn .* (sum(f)/sum(f_Rn))  # normalize
-    return f_Rn
+
+"""
+Apply a gaussian filter to a density field with smoothing radius `R_S`
+making use of wavevectors `kv` using log smoothing
+
+# Arguments:
+- `f::AbstractArray{T,3}`: density field
+- `R_S::T`: filter radius
+- `kv`: wavevector, tuple of (kx, ky, kz)
+
+# Returns:
+- Array{T,3}: log-smoothed density field
+"""
+function smooth_loggauss(f::AbstractArray{T,3}, R_S::T, kv) where {AA, T}
```

Comment:
I optimized this function by performing the operations in-place -- this was probably unnecessary.